spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java

```diff
@@ -141,14 +140,9 @@ private List<ServiceInstance> getNamespaceServiceInstances(EndpointSubsetNS es,
 		List<ServiceInstance> instances = new ArrayList<>();
 
 		Service service = client.services().inNamespace(namespace).withName(serviceId).get();
-		Map<String, String> serviceMetadata = serviceMetadata(serviceId, service, properties, subsets);
+		Map<String, String> serviceMetadata = serviceMetadata(serviceId, service, properties, subsets, namespace);
 
 		for (EndpointSubset endpointSubset : subsets) {
-
-			if (properties.allNamespaces()) {
```

Comment:
this has moved to the method that computes metadata only. But also fixes kind of an inconsistency. We were returning the namespace metadata (via `k8s_namespace` key) only for when all-namespaces was true. But why? the namespace is accessible for all other options as-well, so I decided to return it regardless of this setting.